user-management/client/src/lib/data.ts

```diff
@@ -0,0 +1,71 @@
+export const tokenKey = 'um.token';
+
+export function saveToken(token: string | undefined): void {
+  if (token) {
+    sessionStorage.setItem(tokenKey, token);
+  } else {
+    sessionStorage.removeItem(tokenKey);
+  }
+}
+
+export function readToken(): string {
+  const token = sessionStorage.getItem(tokenKey);
+  if (!token) throw new Error('No token found');
+  return token;
+}
+
+export type UnsavedTodo = {
+  task: string;
+  isCompleted: boolean;
+};
+export type Todo = UnsavedTodo & {
+  todoId: number;
+};
+
+export async function readTodos(): Promise<Todo[]> {
+  const req = {
+    headers: {
+      Authorization: `Bearer ${readToken()}`,
+    },
+  };
+  const res = await fetch('/api/todos', req);
+  if (!res.ok) throw new Error(`fetch Error ${res.status}`);
+  return await res.json();
+}
+
+export async function insertTodo(todo: UnsavedTodo): Promise<Todo> {
+  const req = {
+    method: 'POST',
+    headers: {
+      'Content-Type': 'application/json',
+      Authorization: `Bearer ${readToken()}`,
+    },
+    body: JSON.stringify(todo),
+  };
+  const res = await fetch('/api/todos', req);
+  if (!res.ok) throw new Error(`fetch Error ${res.status}`);
+  return await res.json();
+}
+
+export async function updateTodo(todo: Todo): Promise<Todo> {
+  const req = {
+    method: 'PUT',
+    headers: {
+      'Content-Type': 'application/json',
+      Authorization: `Bearer ${readToken()}`,
+    },
+    body: JSON.stringify(todo),
+  };
+  const res = await fetch(`/api/todos/${todo.todoId}`, req);
+  if (!res.ok) throw new Error(`fetch Error ${res.status}`);
+  return await res.json();
+}
+
+export async function removeTodo(todoId: number): Promise<void> {
+  const req = {
+    method: 'DELETE',
+    Authorization: `Bearer ${readToken()}`,
```

Comment:
You are missing the `headers` field here. Though since this function is never used, you wouldn't ever find the bug!